### DISCUSSION TITLE
如何在画布大小改变后，重新按照新的大小重新渲染呀？

### DISCUSSION THREAD

**Discussion by Lyong0216** [antvis/X6#2905] [Q&A]
我想实现在画布大小尺寸发生变化后，需要依据新的大小重新绘制或按比例缩放，该怎么做呀？有人回答下吗？

**Comment by xiazhaohui** [comment]
> 我想实现在画布大小尺寸发生变化后，需要依据新的大小重新绘制或按比例缩放，该怎么做呀？有人回答下吗？

设置画布自动铺满，你可以监听页面resize事件，及时改变画布的实际渲染尺寸

```
graph.resize(width, height)
```

**Reply by Lyong0216** [reply]
设置宽高只能固定值吗？没办法做适配吗？

**Reply by laobieyi** [reply]
宽度适配可以使用 display:flex 配合子节点的 flex: 1 来完成适配，不能同时适配宽高。
可以参考这个连接：https://x6.antv.vision/zh/docs/api/graph/graph#autoresize

**Comment by lloydzhou** [comment]
不建议监听页面的resize事件。
建议直接监听画布父容器大小，然后调用
```
graph.resize(width, height)
```
可以参考这个实现  https://github.com/lloydzhou/antv-x6-vue/blob/master/src/Graph.tsx#L80 
使用 `resize-detector`这个包监听父容器大小。
